src/module.rs

```diff
@@ -191,18 +191,15 @@ fn ansi_strings_modified(ansi_strings: Vec<ANSIString>, shell: Shell) -> Vec<ANS
                             _ => x.to_string(),
                         }
                     }
-                    MAYBE_ESCAPE_END => {
-                        if escaped {
-                            escaped = false;
-                            match shell {
-                                Shell::Bash => String::from("m\u{5c}\u{5d}"), // => m\]
-                                Shell::Zsh => String::from("m\u{25}\u{7d}"),  // => m%}
-                                _ => x.to_string(),
-                            }
-                        } else {
-                            x.to_string()
+                    MAYBE_ESCAPE_END if escaped => {
+                        escaped = false;
+                        match shell {
+                            Shell::Bash => String::from("m\u{5c}\u{5d}"), // => m\]
+                            Shell::Zsh => String::from("m\u{25}\u{7d}"),  // => m%}
+                            _ => x.to_string(),
                         }
                     }
+                    '$' if Shell::Bash == shell => String::from("\\$"),
```

Comment:
Minor complaint: I'd like to have an explanation for why this is needed. It can either go here or in the block comment at the top of the function--you can even reference the issue number if you want to keep it short. Just something so if someone wants to refer back in the future, they can find some information, e.g. `// Escape $ to avoid `$()` injection attack, see #658 for more`.

Reply:
Yes sorry, you are right. I will add a documentation comment right now.